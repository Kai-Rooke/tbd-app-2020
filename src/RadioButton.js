import React from 'react'

class RadioButton extends React.Component
{
    render()
    {
        return (
                <div>
                    <label for="first">
                    <input type="radio" id= "first" />1st radio button
                    </label>

                    <label for="second">
                    <input type="radio" id= "second" />2nd radio button
                    </label>

                    <label for="third">
                    <input type="radio" id= "third" />3rd radio button
                    </label>
                    
                </div>
        )
    }
}

export default RadioButton;