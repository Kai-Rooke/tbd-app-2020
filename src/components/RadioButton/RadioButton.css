.radio-group input{
    margin-right: 1%;
}

.radio-buttons {
    margin-left: 1.5%;
}