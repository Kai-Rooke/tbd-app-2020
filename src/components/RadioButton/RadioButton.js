import React from 'react'

class RadioButton extends React.Component
{
    render()
    {
        return (
                <div>
                    <label htmlFor="first">
                    <input type="radio" id= "first" />1st radio button
                    </label>

                    <label htmlFor="second">
                    <input type="radio" id= "second" />2nd radio button
                    </label>

                    <label htmlFor="third">
                    <input type="radio" id= "third" />3rd radio button
                    </label>
                    
                </div>
        )
    }
}

export default RadioButton;