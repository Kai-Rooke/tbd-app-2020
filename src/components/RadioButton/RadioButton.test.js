import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import RadioButton from './RadioButton';

//cleans up 
afterEach(cleanup);

var options = [
    {label: "UK National", id: "UK", value: "UK"},
    {label: "EU National", id: "EU", value: "EU"},
    {label: "Other", id: "Other", value: "Other"}
];

it("Renders Radio Button", () => {
    const { asFragment } = render(<RadioButton heading="Nationality" options={options} name="test"/>);
    expect(asFragment()).toMatchSnapshot();
});
