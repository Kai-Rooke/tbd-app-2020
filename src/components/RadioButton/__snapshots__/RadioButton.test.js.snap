// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Radio Button 1`] = `
<DocumentFragment>
  <div
    class="radio-buttons"
  >
    <h4>
      Nationality
    </h4>
    <p />
    <div
      class="radio-group"
    >
      <input
        data-testid="UK"
        id="UK"
        name="test"
        type="radio"
        value="UK"
      />
      <label
        for="UK"
      >
        UK National
      </label>
    </div>
    <div
      class="radio-group"
    >
      <input
        data-testid="EU"
        id="EU"
        name="test"
        type="radio"
        value="EU"
      />
      <label
        for="EU"
      >
        EU National
      </label>
    </div>
    <div
      class="radio-group"
    >
      <input
        data-testid="Other"
        id="Other"
        name="test"
        type="radio"
        value="Other"
      />
      <label
        for="Other"
      >
        Other
      </label>
    </div>
  </div>
</DocumentFragment>
`;
